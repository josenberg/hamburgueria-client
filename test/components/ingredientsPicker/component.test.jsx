import React from 'react';
import renderer from 'react-test-renderer';

import IngredientsPicker from '../../../app/components/ingredientsPicker';

test('IngredientPicker should render without issues', () => {
  const component = renderer.create(
    <IngredientsPicker
      ingredients={[{
        id: 1,
        name: 'alface',
        displayName: 'Alface',
        price: 0.4,
      }]}
      item={{
        ingredients: [{
          id: 1,
          name: 'alface',
          displayName: 'Alface',
          price: 0.4,
          quantity: 1,
        }],
      }}
      updateItem={() => {}}
      rules={[]}
    />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
