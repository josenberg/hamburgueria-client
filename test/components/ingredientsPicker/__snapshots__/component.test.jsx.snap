// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IngredientPicker should render without issues 1`] = `
<div
  className="grid"
  style={
    Object {
      "colGap": "0px",
      "display": "grid",
      "gridTemplateColumns": "1fr auto auto",
      "gridTemplateRows": "auto auto",
      "rowGap": "0px",
    }
  }
>
  <div
    className="grid-cell"
    style={
      Object {
        "display": "grid",
        "gridColumn": "auto",
        "gridRow": "1",
      }
    }
  >
    <h3
      style={
        Object {
          "margin": "0 0 20px 0",
        }
      }
    >
       Escolha os Ingredientes do seu lanche 
    </h3>
  </div>
  <div
    className="grid-cell"
    style={
      Object {
        "display": "grid",
        "gridColumn": "auto",
        "gridRow": "2",
      }
    }
  >
    <div
      className="grid"
      style={
        Object {
          "colGap": "0px",
          "display": "grid",
          "gridTemplateColumns": "auto 40px 40px 40px",
          "gridTemplateRows": "auto",
          "rowGap": "10px",
        }
      }
    >
      <div
        className="grid-cell"
        style={
          Object {
            "display": "grid",
            "gridColumn": "1",
            "gridRow": "1",
          }
        }
      >
        Alface
      </div>
      <div
        className="grid-cell"
        style={
          Object {
            "display": "grid",
            "gridColumn": "2",
            "gridRow": "1",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "background": "none",
              "border": "none",
              "margin": "0",
              "padding": "0",
            }
          }
          type="submit"
        >
          <img
            alt=""
            src="test-file-stub"
          />
        </button>
      </div>
      <div
        className="grid-cell"
        style={
          Object {
            "display": "grid",
            "gridColumn": "3",
            "gridRow": "1",
            "textAlign": "center",
          }
        }
      >
        1
      </div>
      <div
        className="grid-cell"
        style={
          Object {
            "display": "grid",
            "gridColumn": "4",
            "gridRow": "1",
          }
        }
      >
        <button
          onClick={[Function]}
          style={
            Object {
              "background": "none",
              "border": "none",
              "margin": "0",
              "padding": "0",
            }
          }
          type="submit"
        >
          <img
            alt=""
            src="test-file-stub"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="grid-cell"
    style={
      Object {
        "display": "grid",
        "gridColumn": "auto",
        "gridRow": "3",
      }
    }
  >
    <p
      style={
        Object {
          "fontFamily": "Lalezar, cursive",
          "fontSize": "30px",
          "margin": "0",
          "padding": "0 10px 0 0",
          "textAlign": "right",
        }
      }
    >
      R$Â 0,40
    </p>
  </div>
</div>
`;
